actors/builtin/miner/deadline_state_test.go

```diff
@@ -0,0 +1,379 @@
+
+	require.Equal(t, uint64(len(partitionSectors)), partitions.Length())
+
+	expectedDeadlineExpQueue := make(map[abi.ChainEpoch][]uint64)
+	var partitionsWithEarlyTerminations []uint64
+
+	expectPartIndex := int64(0)
+	var partition miner.Partition
+	err = partitions.ForEach(&partition, func(partIdx int64) error {
+		require.Equal(t, expectPartIndex, partIdx)
+		expectPartIndex++
+
+		partSectorNos := partitionSectors[partIdx]
+
+		partFaults, err := bitfield.IntersectBitField(faults, partSectorNos)
+		require.NoError(t, err)
+
+		partRecovering, err := bitfield.IntersectBitField(recovering, partSectorNos)
+		require.NoError(t, err)
+
+		partTerminations, err := bitfield.IntersectBitField(terminations, partSectorNos)
+		require.NoError(t, err)
+
+		unterminatedSectors, err := bitfield.SubtractBitField(partSectorNos, partTerminations)
+		require.NoError(t, err)
+		partSectors := selectSectors(t, sectors, unterminatedSectors)
+
+		assertPartitionState(t,
+			&partition, partSectors, sectorSize, partSectorNos, partFaults, partRecovering, partTerminations)
+
+		earlyTerminated, err := adt.AsArray(store, partition.EarlyTerminated)
+		require.NoError(t, err)
+		if earlyTerminated.Length() > 0 {
+			partitionsWithEarlyTerminations = append(partitionsWithEarlyTerminations, uint64(partIdx))
+		}
+
+		// The partition's expiration queue is already tested by the
+		// partition tests.
+		//
+		// Here, we're making sure it's consistent with the deadline's queue.
+		q, err := adt.AsArray(store, partition.ExpirationsEpochs)
+		require.NoError(t, err)
+		err = q.ForEach(nil, func(epoch int64) error {
+			expectedDeadlineExpQueue[abi.ChainEpoch(epoch)] = append(
+				expectedDeadlineExpQueue[abi.ChainEpoch(epoch)],
+				uint64(partIdx),
+			)
+			return nil
+		})
+		require.NoError(t, err)
+
+		return nil
+	})
+	require.NoError(t, err)
+
+	allSectors, err := bitfield.MultiMerge(partitionSectors...)
+	require.NoError(t, err)
+
+	allSectorsCount, err := allSectors.Count()
+	require.NoError(t, err)
+
+	deadSectorsCount, err := terminations.Count()
+	require.NoError(t, err)
+
+	require.Equal(t, dl.LiveSectors, allSectorsCount-deadSectorsCount)
+	require.Equal(t, dl.TotalSectors, allSectorsCount)
+
+	assertBitfieldsEqual(t, dl.PostSubmissions, posts)
+
+	// Validate expiration queue. The deadline expiration queue is a
+	// superset of the partition expiration queues because we never remove
+	// from it.
+	{
+		expirationEpochs, err := adt.AsArray(store, dl.ExpirationsEpochs)
+		require.NoError(t, err)
+		for epoch, partitions := range expectedDeadlineExpQueue {
+			var bf abi.BitField
+			found, err := expirationEpochs.Get(uint64(epoch), &bf)
+			require.NoError(t, err)
+			require.True(t, found)
+			for _, p := range partitions {
+				present, err := bf.IsSet(p)
+				require.NoError(t, err)
+				assert.True(t, present)
+			}
+		}
+	}
+
+	// Validate early terminations.
+	assertBitfieldEquals(t, dl.EarlyTerminations, partitionsWithEarlyTerminations...)
```

Comment:
As suggested in #740, extract parts here that represent invariant checking, un-parameterized, perhaps to a method on the deadline state object. We can use that in other tests, including actor tests, to gain greater scenario coverage.

Reply:
Done. It does take the set of all sectors in the system, but I figure that's something all tests can pass around.